tokio-util/src/time/delay_queue.rs

```diff
@@ -867,33 +1138,35 @@ impl<T> wheel::Stack for Stack<T> {
             let mut contains = false;
 
             while let Some(idx) = next {
+                let data = &store[Key::new(idx)];
+
                 if idx == *item {
                     debug_assert!(!contains);
                     contains = true;
                 }
 
-                next = store[idx].next;
+                next = data.next;
             }
 
             contains
         });
 
-        if let Some(next) = store[*item].next {
-            store[next].prev = store[*item].prev;
+        if let Some(next) = store[key].next {
+            store[Key::new(next)].prev = store[key].prev;
```

Comment:
Seems like the `next`/`prev` fields and variables should just be of type `Key` themselves instead of wrapping them here. We want as little wrapping and unwrapping as possible.